middle_end/flambda2/to_cmm/to_cmm_shared.ml

```diff
@@ -224,6 +232,12 @@ let const_static cst =
     (* Just in case of future big endian support, this is also written
        explicitly in two halves. *)
     [cint32 i; cint32 0l]
+  | Naked_int16 i ->
+    (* In keeping with Naked_float32, this is padded to 8 bytes, but it's not
+       obvious to me why we are doing this instead of requiring alignment on
+       other values that need it. *)
```

Comment:
I think it's less a question of other values requiring some kind of alignment, and more the fact that the assumption that most things take exactly one memory word is baked in so many places (well, except for cmm static consts, but they must be compatible with all of the rest of the code), that it's easier to be explicit here. I think it'll be easier when we want to try packing: we'll only have to remove the explicit padding that is inserted here.